frontend/src/MyProfile/ProfileModal.jsx

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useContext } from 'react';
+import './ProfileModal.css';
+import { UserContext } from '../UserContext';
+
+export default function ProfileModal({ isOpen, onClose, handleSubmitPatientInfo }) {
+  const { user } = useContext(UserContext);
+  const [formData, setFormData] = useState({
+    firstname: '',
+    lastname: '',
+    place_of_birth: '',
+    date_of_birth: '',
+    sex: '',
+    height: '',
+    weight: '',
+    occupation: '',
+    address: '',
+    phone: '',
+    complaint: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setFormData((prevData) => ({ ...prevData, [name]: value }));
+  };
+
+  if (!isOpen) return null;
+
+  return (
+    <div className="modal-overlay">
+      <div className="modal-content">
+        <form onSubmit={(e) => handleSubmitPatientInfo(e, formData)}>
+          <p>CREATE PATIENT PROFILE</p>
+          <input name="firstname" placeholder="First Name" value={formData.firstname} onChange={handleChange} required />
+          <input name="lastname" placeholder="Last Name" value={formData.lastname} onChange={handleChange} required />
+          <input name="place_of_birth" placeholder="Place of Birth" value={formData.place_of_birth} onChange={handleChange} required />
+          <input name="date_of_birth" placeholder="Date of Birth" type="date" value={formData.date_of_birth} onChange={handleChange} required />
+          <input name="sex" placeholder="Sex" value={formData.sex} onChange={handleChange} required />
+          <input name="height" placeholder="Height (in)" type="number" value={formData.height} onChange={handleChange} required />
+          <input name="weight" placeholder="Weight (lbs)" type="number" value={formData.weight} onChange={handleChange} required />
+          <input name="occupation" placeholder="Occupation" value={formData.occupation} onChange={handleChange} required />
+          <input name="address" placeholder="Address" value={formData.address} onChange={handleChange} required />
+          <input name="phone" placeholder="Phone" type="tel" value={formData.phone} onChange={handleChange} required />
+          <textarea name="complaint" placeholder="Complaint" value={formData.complaint} onChange={handleChange} />
```

Comment:
could we specify which inputs are optional and which are mandatory? for example, i'm assuming complaint is optional when a user is first creating their profile, whereas first and last name probably aren't

Reply:
Yes, I specified this in the schema.prisma, and it doesn't cause any issue on the frontend. Should I still specify it?

Reply:
yes! from a user standpoint, it would definitely be helpful (if not expected) to know what input is mandatory and what is optional so that i don't run into any unexpected errors when i'm creating my profile.